controllers_v2/keysend.ctrl.go

```diff
@@ -149,6 +160,26 @@ func (controller *KeySendController) MultiKeySend(c echo.Context) error {
 	return c.JSON(status, result)
 }
 
+func (controller *KeySendController) checkKeysendPaymentAllowed(ctx context.Context, amount, userID int64) (resp *responses.ErrorResponse) {
+	syntheticPayReq := &lnd.LNPayReq{
+		PayReq: &lnrpc.PayReq{
+			NumSatoshis: amount,
+		},
+		Keysend: true,
+	}
+	resp, err := controller.svc.CheckPaymentAllowed(ctx, syntheticPayReq, userID)
+	if resp != nil {
+		controller.svc.Logger.Errorf("User does not have enough balance user_id:%v amount:%v", userID, syntheticPayReq.PayReq.NumSatoshis)
+		return resp
+	}
+	if err != nil {
+		errResp := responses.GeneralServerError
```

Comment:
Check the error first, and log the error and just return a GeneralServerError here.

Reply:
Done!